src/base/utils/net.cpp

```diff
@@ -91,6 +92,37 @@ namespace Utils
             return subnet.first.toString() + '/' + QString::number(subnet.second);
         }
 
+        QHostAddress canonicalIPv6Addr(const QHostAddress &addr)
+        {
+            // Link-local IPv6 textual address always contains a scope id (or zone index)
+            // The scope id is appended to the IPv6 address using the '%' character
+            // The scope id can be either a interface name or an interface number
+            // Examples:
+            // fe80::1%ethernet_17
+            // fe80::1%13
+            // The interface number is the mandatory supported way
+            // Unfortunately for us QHostAddress::toString() outputs (at least on Windows)
+            // the interface name, and libtorrent/boost.asio only support an interface number
+            // as scope id. Furthermore, QHostAddress doesn't have any convenient method to
+            // affect this, so we jump through hoops here.
+            if (addr.protocol() != QAbstractSocket::IPv6Protocol)
+                return {};
```

Comment:
If applying this function to IPv4 address isn't valid "use case" you should prevent it in any way (at least add an appropriate assertion). Otherwise it may allow logically incorrect code.

Reply:
I am confused. How do I prevent it in any way?
And won't the assertion work only in debug?

Reply:
>How do I prevent it in any way?


You can throw RuntimeError...

Or you can modify it to be valid for any IP (and rename it to `canonicalIPAddr`).

Reply:
Or even it can return IPv6 representstion of IPv4... unless I am wrong it is possible.

Reply:
[QHostAddress::toIPv6Address()](https://doc.qt.io/qt-5/qhostaddress.html#toIPv6Address) seems to do the job. So it becomes:
```c++
if (addr.protocol() != QAbstractSocket::IPv6Protocol)
                return {addr.toIPv6Address()};
```

Reply:
And also apply all remaining modifications.

Reply:
Or it already in "canonical" form?

Reply:
>Or it already in "canonical" form?

It can't be, because at this point it is IPv4.
If you mean below, there is no way to know beforehand. So we convert unconditionally.

Reply:
I would implement anothet variant instead. I mean `canonicalIPAddr` that returns IP of the same type as its argument.
BTW, we can use it unconditionally since it always returns valid result.